src/lib/yang/binary.lua

```diff
@@ -495,6 +536,7 @@ function selftest()
       assert(routing_table:lookup_ptr(key).value.port == 10)
       key.addr = util.ipv4_pton('3.4.5.6')
       assert(routing_table:lookup_ptr(key).value.port == 2)
+      assert(data.next_hop.ipv4 == util.ipv4_pton('5.6.7.8'))
```

Comment:
I've fixed this. It should now print out the round (FWIW it fails on the second round which is when it's been written to binary and reloaded). When you take a look `data.next_hop` ends up just being `{}` when loaded from binary.